src/ICal/ICal.php

```diff
@@ -1236,14 +1235,18 @@ protected function processRecurrences()
                 // Tag as generated by a recurrence rule
                 $anEvent['RRULE_array'][2] = self::RECURRENCE_EVENT;
 
+                $countNb = 0;
+
                 $isAllDayEvent = (strlen($anEvent['DTSTART_array'][1]) === 8) ? true : false;
 
                 $initialStart             = new \DateTime($anEvent['DTSTART_array'][1]);
                 $initialStartTimeZoneName = $initialStart->getTimezone()->getName();
+                $endTimestamp             = null;
 
                 if (isset($anEvent['DTEND'])) {
                     $initialEnd             = new \DateTime($anEvent['DTEND_array'][1]);
                     $initialEndTimeZoneName = $initialEnd->getTimezone()->getName();
+                    $endTimestamp = $initialEnd->getTimestamp();
```

Comment:
Don't see the need for this

Reply:
This is how the end time stamp is set when specified explicitly in the event's DTEND. I don't see where it would be set below. Maybe this is in regards to your "swap" comment?